src/reactviews/pages/SchemaDesigner/graph/SchemaDiagramFlow.tsx

```diff
@@ -230,12 +253,59 @@ export const SchemaDesignerFlow = () => {
                 onNodeDragStop={() => {
                     eventBus.emit("pushState");
                 }}
+                onBeforeDelete={async (props) => {
+                    if (props.nodes.length === 0 && props.edges.length === 0) {
+                        return true;
+                    }
+                    return await deleteElementsConfirmation();
+                }}
                 minZoom={0.05}
                 fitView>
                 <Controls />
                 <MiniMap pannable zoomable />
                 <Background variant={BackgroundVariant.Dots} gap={12} size={1} />
             </ReactFlow>
+            <Dialog
+                open={open}
+                onOpenChange={(_event, data) => {
+                    setOpen(data.open);
+                }}>
+                <DialogSurface>
+                    <DialogBody>
+                        <DialogTitle>{locConstants.schemaDesigner.deleteConfirmation}</DialogTitle>
+                        <DialogContent>
+                            {locConstants.schemaDesigner.deleteConfirmationContent}
+                        </DialogContent>
+
+                        <DialogActions>
+                            <DialogTrigger disableButtonEnhancement>
+                                <Button
+                                    appearance="primary"
```

Comment:
should we make `cancel` primary for both confirmation dialogs?

Reply:
Chatted with Aasim offline, the original implementation is consistent with the Fluent guideline. https://react.fluentui.dev/iframe.html?viewMode=docs&id=components-dialog--docs#confirmation